components/overview/InvestMoreModal.js

```diff
@@ -368,16 +379,29 @@ const InvestMoreModal = withSlideIn(
 															</Popover>
 														</span>
 													</div>
+												) : amount >
+												  balance?.availableBalance /
+														Math.pow(10, networkInfo.decimalPlaces) -
+														networkInfo.reserveAmount ? (
+													<div className="rounded-lg px-5 py-2 text-sm bg-red-200 text-red-600 mb-4">
+														<span>
+															Insufficient Balance: We cannot stake this amount
+															as it crosses the maximum available balance for
+															staking in your account.{" "}
+														</span>
+													</div>
 												) : (
-													amount >
-														balance?.availableBalance /
-															Math.pow(10, networkInfo.decimalPlaces) -
-															networkInfo.reserveAmount && (
+													controllerAccount &&
+													controllerBalances &&
+													transactionFee +
+														apiInstance?.consts.balances.existentialDeposit.toNumber() >
```

Comment:
change this to `transactionFee + (apiInstance?.consts.balances.existentialDeposit.toNumber() / 2)`

Reply:
@sahilnanda1995 i do not understand this. if a user is using a separate controller then that account will get reaped if its balance drops below the existential deposit.

Reply:
> then that account will get reaped if its balance drops below the existential deposit
hmm right, i was somehow thinking of including a total balance check for an account but that will be low yield

Can you also crosscheck this and push a fix for the same for staking, reward-calculator and validators page module

Reply:
@sahilnanda1995 I think its already being checked in validator and reward-calculator page but I will check them.